test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -1282,6 +1282,169 @@ public void TryGetLink_WithCustomAddress_CanGenerateLink_RespectsLinkOptions_Sup
+
+            // Act1
+            var template = linkGenerator.GetTemplate(values: new { controller = "Product", action = "Edit" });
+
+            // Assert1
+            Assert.NotNull(template);
+
+            // Act2
+            var link = template.MakeUrl(new { id = 10 });
+
+            // Assert2
+            Assert.Equal("/Product/Edit/10", link);
+
+            // Act3
+            link = template.MakeUrl(new { id = 25 });
+
+            // Assert3
+            Assert.Equal("/Product/Edit/25", link);
+        }
+
+        [Fact]
+        public void GetTemplate_ByRouteName_ReturnsTemplate()
+        {
+            // Arrange
+            var endpoint1 = EndpointFactory.CreateMatcherEndpoint(
+                "Product/Edit/{id}",
+                requiredValues: new { controller = "Product", action = "Edit", area = (string)null, page = (string)null },
+                defaults: new { controller = "Product", action = "Edit", area = (string)null, page = (string)null },
+                metadata: new RouteNameMetadata("EditProduct"));
+            var linkGenerator = CreateLinkGenerator(endpoint1);
+
+            // Act1
+            var template = linkGenerator.GetTemplate("EditProduct", values: new { });
+
+            // Assert1
+            Assert.NotNull(template);
+
+            // Act2
+            var link = template.MakeUrl(new { id = 10 });
+
+            // Assert2
+            Assert.Equal("/Product/Edit/10", link);
+
+            // Act3
+            link = template.MakeUrl(new { id = 25 });
+
+            // Assert3
+            Assert.Equal("/Product/Edit/25", link);
+        }
+
+        [Fact]
+        public void GetTemplate_ByCustomAddress_ReturnsTemplate()
+        {
+            // Arrange
+            var services = GetBasicServices();
+            services.TryAddEnumerable(
+                ServiceDescriptor.Singleton<IEndpointFinder<INameMetadata>, EndpointFinderByName>());
+            var endpoint1 = EndpointFactory.CreateMatcherEndpoint(
+                "Product/Edit/{id}",
+                requiredValues: new { controller = "Product", action = "Edit", area = (string)null, page = (string)null },
+                defaults: new { controller = "Product", action = "Edit", area = (string)null, page = (string)null });
+            var endpoint2 = EndpointFactory.CreateMatcherEndpoint(
+                "Customers/Details/{id}",
+                requiredValues: new { controller = "Customers", action = "Details" },
+                defaults: new { controller = "Customers", action = "Details" },
+                metadata: new NameMetadata("CustomerDetails"));
+            var linkGenerator = CreateLinkGenerator(new[] { endpoint1, endpoint2 }, new RouteOptions(), services);
+
+            // Act1
+            var template = linkGenerator.GetTemplateByAddress<INameMetadata>(new NameMetadata("CustomerDetails"));
+
+            // Assert1
+            Assert.NotNull(template);
```

Comment:
It would be much better for these kinds of tests if the test was focused on the fact that a template object was returned, ie: don't generate a URL, just inspect the details of the template.

Reply:
Sure, I agree.